zeek/bin/spicyz.cc

```diff
@@ -0,0 +1,226 @@
+        }
+    }
+
+    while ( optind < argc )
+        driver_options->inputs.emplace_back(argv[optind++]);
+
+    if ( driver_options->inputs.empty() )
+        return hilti::result::Error("no input file given");
+
+    if ( ! compiler_options->debug ) {
+        if ( compiler_options->debug_trace || compiler_options->debug_flow )
+            return hilti::result::Error("cannot use --optimize with --cgdebug");
+    }
+
+    if ( driver_options->output_path.empty() )
+        return hilti::result::Error("no output file for object code given, use -o <file>.hlto");
+
+    if ( ! util::endsWith(driver_options->output_path, ".hlto") )
+        return hilti::result::Error("output file must have '.hlto' extension");
+
+    return Nothing();
+}
+
+int main(int argc, char** argv) {
+    spicy::zeek::Driver driver;
+
+    hilti::Options compiler_options;
+    hilti::driver::Options driver_options;
+
+    driver_options.execute_code = true;
+    driver_options.include_linker = true;
+
+    compiler_options.cxx_include_paths = {spicy::zeek::configuration::CxxZeekIncludeDirectory,
+                                          spicy::zeek::configuration::CxxBrokerIncludeDirectory};
+
+    if ( hilti::configuration().uses_build_directory ) {
+        compiler_options.cxx_include_paths.emplace_back(spicy::zeek::configuration::CxxRuntimeIncludeDirectoryBuild);
+        compiler_options.cxx_include_paths.emplace_back(spicy::zeek::configuration::CxxAutogenIncludeDirectoryBuild);
+        compiler_options.library_paths.emplace_back(spicy::zeek::configuration::PluginLibraryDirectoryBuild);
+    }
+    else {
+        compiler_options.cxx_include_paths.emplace_back(
+            spicy::zeek::configuration::CxxRuntimeIncludeDirectoryInstallation);
+        compiler_options.library_paths.emplace_back(spicy::zeek::configuration::PluginLibraryDirectoryInstallation);
+    }
+
+    if ( auto path = getenv("ZEEK_SPICY_PATH") ) {
+        for ( const auto& dir : hilti::rt::split(path, ":") )
+            compiler_options.library_paths.emplace_back(dir);
+    }
+
+    if ( auto rc = parseOptions(argc, argv, &driver_options, &compiler_options); ! rc ) {
+        hilti::logger().error(rc.error().description());
+        exit(1);
+    }
+
+    if ( driver_options.output_cxx )
+        driver_options.execute_code = false;
+
+#ifndef NDEBUG
+    ZEEK_DEBUG("Search paths:");
+
+    for ( const auto& x : compiler_options.library_paths ) {
+        ZEEK_DEBUG(hilti::rt::fmt("  %s", x.native()));
+    }
+#endif
+
+    driver.setDriverOptions(std::move(driver_options));
+    driver.setCompilerOptions(std::move(compiler_options));
+
+    driver.initialize();
+
+    for ( auto p : driver.driverOptions().inputs ) {
+        if ( ! driver.loadFile(p) )
+            exit(1); // error already reported
+    }
+
+    if ( ! driver.compile() )
+        exit(1); // error already reported
+
+    exit(0);
```

Comment:
Redundant. If at all let's `return 0;`, but probably better to just rely on the default behavior of `main`.